apps/ftu/js/ui.js

```diff
@@ -149,13 +150,22 @@ var UIManager = {
     this.hiddenWifiSecurity.addEventListener('change', this);
     this.wifiJoinButton.disabled = true;
 
-    this.hiddenWifiPassword.addEventListener('keyup', function() {
-      this.wifiJoinButton.disabled = !WifiHelper.isValidInput(
-        this.hiddenWifiSecurity.value,
-        this.hiddenWifiPassword.value,
-        this.hiddenWifiIdentity.value
+    var checkHiddenWifiJoin = function() {
```

Comment:
Why not declare and bind the function here, instead of binding separately for each event listener below?